packages/core/src/Config.cpp

```diff
@@ -707,15 +1128,229 @@ bool Config::configure_services(const SystemHandleInfoMap& info_map) const
+            }
+          };
+
+      if (client_info.reply_type.empty())
+      {
+        valid &= it->second.service_client->create_client_proxy(
+          client_info.name,
+          //*client_type,
+          (client_info.type.find(".") == std::string::npos
+                     ? *client_type
+                     : *m_types.at(client_info.type.substr(0, client_info.type.find(".")))),
+          callback,
+          config_or_empty_node(client, config.middleware_configs));
+      }
+      else
+      {
+        const xtypes::DynamicType* client_reply_type = resolve_type(it->second.types, client_info.reply_type);
+
+        valid &= it->second.service_client->create_client_proxy(
+          client_info.name,
+          //*client_type,
+          (client_info.type.find(".") == std::string::npos
+                     ? *client_type
+                     : *m_types.at(client_info.type.substr(0, client_info.type.find(".")))),
+          (client_info.reply_type.find(".") == std::string::npos
+                     ? *client_reply_type
+                     : *m_types.at(client_info.reply_type.substr(0, client_info.reply_type.find(".")))),
+          callback,
+          config_or_empty_node(client, config.middleware_configs));
+      }
+    }
+  }
+
+  return valid;
+}
+
+//==============================================================================
+bool Config::check_topic_compatibility(
+    const SystemHandleInfoMap& info_map,
+    const std::string& topic_name,
+    const TopicConfig& config) const
+{
+  bool valid = true;
+  for (const std::string& from : config.route.from)
+  {
+    const auto it_from = info_map.find(from);
+    TopicInfo topic_info_from = remap_if_needed(from, config.remap, {topic_name, config.message_type, ""});
+    const xtypes::DynamicType* from_type = resolve_type(it_from->second.types, topic_info_from.type);
+    //const auto from_type = it_from->second.types.find(topic_info_from.type);
+    /*
+    if (from_type == it_from->second.types.end())
+    {
+      std::cerr << "Type [" << topic_info_from.type << "] not defined in middleware ["
+                << it_from->first << "]" << std::endl;
+      valid = false;
+      continue;
+    }
+    */
+
+    for (const std::string& to : config.route.to)
+    {
+      const auto it_to = info_map.find(to);
+      TopicInfo topic_info_to = remap_if_needed(to, config.remap, {topic_name, config.message_type, ""});
+      const xtypes::DynamicType* to_type = resolve_type(it_from->second.types, topic_info_from.type);
+      /*
+      const auto to_type = it_to->second.types.find(topic_info_to.type);
+      if (to_type == it_to->second.types.end())
+      {
+        std::cerr << "Type [" << topic_info_to.type << "] not defined in middleware ["
+                  << it_to->first << "]" << std::endl;
+        valid = false;
+        continue;
+      }
+      */
+
+      //xtypes::TypeConsistency consistency = from_type->second->is_compatible(*to_type->second);
```

Comment:
```suggestion

    for (const std::string& to : config.route.to)
    {
      const auto it_to = info_map.find(to);
      TopicInfo topic_info_to = remap_if_needed(to, 
                                                config.remap, 
                                                {topic_name, 
                                                 config.message_type, ""});
      const xtypes::DynamicType* to_type = resolve_type(it_from->second.types,
                                                        topic_info_from.type);
```
Shall we remove the commented code?